src/crimson/osd/pg_backend.cc

```diff
@@ -1807,3 +1807,40 @@ void PGBackend::on_activate_complete() {
   peering.reset();
 }
 
+std::optional<hobject_t> PGBackend::resolve_oid(
+  const SnapSet &ss,
+  const hobject_t &oid)
```

Comment:
It's not obvious to me why this is desirable.  The intention of the PGBackend interface is to abstract out the difference between a replicated pg and an erasure coded pg.  Mapping a snapshot to a specific clone, however, works identically for replicated and ec pgs.  Why do we want to move this?

Reply:
Probably you moved it because you are passing a PGBackend to ObjectContextLoader.  This doesn't appear to have any actual dependencies on PG or PGBackend members -- it's just a function from (ss, oid) to oid.  This could probably be a free function in defined in object_context.h/cc.